### PR TITLE
[fix #19870] correctly convert index to style category

### PR DESCRIPTION
**Pull request by 3nids**
the shift is already done in the list, so no need to hack

### CODE REVIEW

**Comment by 3nids** [comment]
@elpaso, still building to test on my side but I think that should save it.
Sorry for the crash!

**Comment by elpaso** [comment]
Thanks @3nids! Can we remove the `index2category` and directly retrieve the category by index now?

**Comment by 3nids** [comment]
done

File: src/app/qgsmaplayerstylecategoriesmodel.cpp [hunk 1]
```diff
@@ -68,7 +63,16 @@ QVariant QgsMapLayerStyleCategoriesModel::data( const QModelIndex &index, int ro
   if ( !index.isValid() || index.row() >= rowCount() )
     return QVariant();
 
-  QgsMapLayer::StyleCategory category = index2category( index );
+  QgsMapLayer::StyleCategory category = mCategoryList.at( index.row() + ( mShowAllCategories ? 0 : 1 ) );
```
Comment (3nids) [inline src/app/qgsmaplayerstylecategoriesmodel.cpp]: @elpaso sorry, you were right!!!!!

Comment (3nids) [reply src/app/qgsmaplayerstylecategoriesmodel.cpp]: I kept your idea of removing index2category and embed this in data. ok for you?

Comment (elpaso) [reply src/app/qgsmaplayerstylecategoriesmodel.cpp]: Yes, thanks for fixing this!

**Pull request merged**